### ISSUE TITLE
Fix osso-pdf-viewer launcher and other compiler warnings

### ISSUE THREAD

**Issue by MerlijnWajer** [maemo-leste/bugtracker#280]
I've done an initial round of importing and fixing, application now launches and works.

maemo-launcher support doesn't work, the resulting binary has no `main`, and that causes issues for maemo-launcher. There may be more going on, but so I've disabled the launcher, for now.

There are a few more implicit function warnings in gtk, and in general more cleanups that could be done.

There is also a launcher-wip branch.

https://github.com/maemo-leste/osso-pdf-viewer

**Comment by MerlijnWajer** [comment]
Also, in my VM the file chooser was working fine, but it's not on the N900, for some reason.

**Comment by MerlijnWajer** [comment]
@android-808 - I think we had the same issue in DrNokSnes. I think I just traced it.

The environment variable `MYDOCSDIR` has some effect. Try this, for example:

```
MYDOCSDIR=/root/MyDocs osso_pdfviewer
```

Then the dialog works. When it's empty or set to `/home/user/MyDocs`, the dialog doesn't work.

**Comment by MerlijnWajer** [comment]
Unsetting it has the same effect as setting it to an non-existing path or path that can't be accessed because of priv issues.

**Comment by android-808** [comment]
https://github.com/community-ssu/osso-product-info/blob/master/libossoproductinfo.c sets up OSSO_PRODUCT_HARDWARE, which is used by https://github.com/maemo-leste/osso-af-startup/blob/master/services/af-defines.sh to detect if we have an internal MMC. Dependant on MMC, we then set value of MYDOCSDIR.

The OSSO_PRODUCT_HARDWARE part I'm not 100% how it gets the value.

This could be helpful reworking that missing /proc entry for ScummVM.

**Comment by MerlijnWajer** [comment]
I think that if MYDOCSDIR is set correctly (e.g. `/home/user/MyDocs`), then the dialog still doesn't show anything. So there's something else wrong in `libhildonfm`, not just what sets MYDOCSDIR. Try to unset it and run drnoksnes, I bet you you will be able to just  browse and select files. (or set it to something invalid like `/doesnotexist`.

**Comment by MerlijnWajer** [comment]
The latest `osso-af-startup` will only set MYDOCSDIR when `/home/user/MyDocs` is a mount point. So that should take care of the problem in drnoksnes, osso_pdfreader, etc.

**Comment by MerlijnWajer** [comment]
```
17:55 < Wizzup> something with main not found?
17:56 < spinal84> Wizzup, yes
17:56 < spinal84> https://github.com/maemo-leste/hildon-desktop-n900/commit/c0b403ea41bc2fa3b0d2ab9c59800e50be5c33da
17:56 < Wizzup> tmlind: sorry for not jumping straight on the new pvr stuff on the n900, I have so much on my plate that I'm losing track so I need
                to back off a little bit and focus on some parts first
17:56 < Wizzup> looks like exactly the same problem I had on the pdf reader
17:56 < Wizzup> https://github.com/maemo-leste/bugtracker/issues/280
17:56 < spinal84> This flag is needed to make linker option -version-script to work: -export-dynamic
```

**Comment by spinal84** [comment]
Updated launcher-wip branch

**Comment by MerlijnWajer** [comment]
Fixed in 1.9.1, thanks.
